compiler/plugins/target/MetalSPIRV/BUILD.bazel

```diff
@@ -12,12 +12,9 @@ package(
     licenses = ["notice"],  # Apache 2.0
 )
 
-iree_cmake_extra_content(
-    content = """
-if(NOT IREE_TARGET_BACKEND_METAL_SPIRV)
-  return()
-endif()
-""",
+iree_compiler_register_plugin(
+    plugin_id = "hal_target_metal_spirv",
```

Comment:
I looked at moving some of the spirv_cross deps/setup into the plugin too, but I'm not sure how I want to handle Bazel support...

https://github.com/openxla/iree/blob/a8061b02030223ffd842a4eefc6348695179c8ec/CMakeLists.txt#L924-L928 https://github.com/openxla/iree/blob/a8061b02030223ffd842a4eefc6348695179c8ec/build_tools/cmake/iree_external_cmake_options.cmake#L29-L43
https://github.com/openxla/iree/blob/main/build_tools/third_party/spirv_cross/BUILD.overlay
https://github.com/openxla/iree/blob/a8061b02030223ffd842a4eefc6348695179c8ec/build_tools/bazel/workspace.bzl#L135-L140